### PR TITLE
Add error message for invalid concurrency value

### PR DESCRIPTION
**Pull request by mtreinish**
This commit adds a detailed error message when a user passes in a
concurrency outside the expected range (which is >= 0). Previously
we'd raise an unhandled IndexError because of going outside the list
bounds deeper in the code, which wasn't super useful or informative for
users. This fixes that by detecting when we have an invalid input up
front and print that out before we run anything.

Related to: #202

### CODE REVIEW

**Comment by mtreinish** [comment]
This is what I was talking about in #202 it's independent of whatever we decide to do there for the new feature.

**Comment by mtreinish** [comment]
Hmm, that's an odd failure I've not seen before: https://ci.appveyor.com/project/mtreinish/stestr/builds/21100814/job/kf7wbkoc7ug23xoo#L164 I'm going to restart that job. But it might be something we need to watch in the future.

**Comment by mtreinish** [comment]
@masayukig do you have any thoughts on this?

File: stestr/commands/run.py [hunk 1]
```diff
@@ -308,6 +308,11 @@ def run_command(config='.stestr.conf', repo_type='file',
             exit(1)
         repo = util.get_repo_initialise(repo_type, repo_url)
     combine_id = None
+    if concurrency and concurrency < 0:
+        msg = ("The provided concurrency value: %s is not valid. An integer "
+               ">= 0 must be used.\n")
```
Comment (masayukig) [inline stestr/commands/run.py]: heh, 
 `">= 0 must be used.\n" % concurrency)` 
?

File: stestr/commands/run.py [hunk 2]
```diff
@@ -308,6 +308,11 @@ def run_command(config='.stestr.conf', repo_type='file',
...
+        stdout.write(msg)
+        return 2
```
Comment (masayukig) [inline stestr/commands/run.py]: And, if we run stestr with `--slowest` option, slowest test info are showed in this case. So, should we check this return value there?
https://github.com/mtreinish/stestr/pull/208/files#diff-e77977ac6a427e77fc880ed584215c86R197

File: stestr/commands/run.py (hunk 1 above)
Comment (mtreinish) [reply stestr/commands/run.py]: Heh, oops I'll fix that now

File: stestr/commands/run.py (hunk 2 above)
Comment (mtreinish) [reply stestr/commands/run.py]: Hmm, this is an interesting case. Since the return code is non-zero for expected error conditions (like a test failure) where we probably still want to print the slowest tests (when `--slowest` is set). 

I agree that we don't want to print the slowest in the case like this with an invalid input and fails before we run any tests, but I'm not sure of a way to perfectly detect that. Were you thinking check for the unique return code `2` here? I don't think all of these edge conditions will have unique return codes to detect.

Comment (masayukig) [reply stestr/commands/run.py]: well, yeah. I see. So maybe, it's better to check it in the earlier, like after store the concurrency value?
https://github.com/mtreinish/stestr/pull/208/files#diff-e77977ac6a427e77fc880ed584215c86R172

Comment (mtreinish) [reply stestr/commands/run.py]: Ah, yeah that's a good call I'll copy the check there too. We need to keep this inside the method too for users using the python API instead of the CLI. But, we'll never reach the inner call via the cli if the input is bad so it'll avoid the issue you're outlining here.
